sql/catalyst/src/test/scala/org/apache/spark/sql/connector/catalog/InMemoryBaseTable.scala

```diff
@@ -648,7 +648,7 @@ case class PartitionInternalRow(keys: Array[Any])
       return false
     }
     // Just compare by reference, not by value
-    this.keys == other.asInstanceOf[PartitionInternalRow].keys
+    this.keys sameElements other.asInstanceOf[PartitionInternalRow].keys
```

Comment:
This is to fix the 'non-clustered distribution: V2 function with multiple args' test below, which now needs to compare the actual partitionRow values.